client/server/bundler/index.js

```diff
@@ -110,26 +110,7 @@ function middleware( app ) {
 	}
 
 	app.use( waitForCompiler );
-	app.use(
-		webpackMiddleware( compiler, {
-			mode: 'development',
-			publicPath: `/calypso/${ process.env.DEV_TARGET || 'evergreen' }/`,
-			stats: {
-				colors: true,
-				hash: true,
-				version: false,
-				timings: true,
-				assets: false,
-				chunks: false,
-				modules: false,
-				cached: false,
-				reasons: false,
-				source: false,
-				errorDetails: true,
-				entrypoints: false,
-			},
-		} )
-	);
+	app.use( webpackMiddleware( compiler ) );
```

Comment:
https://github.com/webpack/webpack-dev-middleware/blob/master/CHANGELOG.md#breaking-changes

`stats` was removed, `mode` was not a thing.
The value from these three properties is taken from webpack config, don't need to repeat it here.

`stats` and `publicPath` are removed and taken from webpack config. `mode` was never a thing